alerta/app/database/mongo.py

```diff
@@ -49,8 +50,30 @@ def connect(self):
 
         mongo_uri = mongo_uri or app.config['MONGO_URI']  # use app config if no env var overrides
 
+        # Validate SSL options
+        def try_find(k):
+            try:
+                return (os.environ.get(k) or app.config[k])
+            except KeyError: pass
+
+        to_verify = ['MONGO_SSL_CERT', 'MONGO_SSL_CA']
+        ssl_opts = { k: try_find(k) for k in to_verify }
+        missing_ssl_opts = { k:v for k, v in ssl_opts.iteritems() if v is None }
```

Comment:
The python 3x issue I can address, I suspect it's just s/iteritems/items/. 

As for this being overly complicated, that's your call. I didn't know that python had .get(x, False) for defaulting, so I can remove the helper function which certainly reduces perceived complexity.

1. 

If you want to go with your explicit version, I'd observe: (P ∧ ¬Q) ∨ (Q ∧ ¬P) == P ⊻ Q. So, it's just an xor which I find less verbose. The reason I didn't do this boolean statement was that the user doesn't know which variable is missing, and the solution is now specific to two arguments. As a general rule, if I have to deal with multiple elements I handle them as n elements and print errors nicely.

This code can later be pulled out into a helper to handle the general case of validating groups of n config items, and we aren't manually querying the underlying sources of configuration in multiple ways with different strings, all of which could be subtly wrong and will require manual replacement in multiple places if refactored.

Up to you. Shall I rewrite this with the explicit boolean check?